### ISSUE TITLE
UTF-8 chars in console.log

### ISSUE THREAD

**Issue by ilearnio** [arangodb/arangodb#1936]
When I'm trying to log something in Russian language, e.g:

```js
console.log('Привет')
```

I'm getting `\xD0\x9F\xD1\x80\xD0\xB8\xD0\xB2\xD0\xB5\xD1\x82` as the output from `arangod`. It would go fine if at least it would be a valid hexadecimal string, but it's not. When I'm putting in to Chrome's console (it usually decrypts hex to normal strings) I'm getting this:

```
> "\xD0\x9F\xD1\x80\xD0\xB8\xD0\xB2\xD0\xB5\xD1\x82"
"ÐÑÐ¸Ð²ÐµÑ"
```

So I guess it's a bug in arangod/foxx

**Comment by ilearnio** [comment]
Or I might be wrong. It appears that it is valid, the Chrome just can't convert it. It would be nice to see human readable output in the terminal though

**Comment by jsteemann** [comment]
Looking at the code, the escaping seems intentional. All characters that fall out of the ASCII range (this includes all Cyrillic characters) and non-printable ASCII characters get escaped.
The following patch will fix that:
```diff
diff --git a/lib/Basics/tri-strings.cpp b/lib/Basics/tri-strings.cpp
index 5bf86cb..2f99044 100644
--- a/lib/Basics/tri-strings.cpp
+++ b/lib/Basics/tri-strings.cpp
@@ -932,11 +932,16 @@ char* TRI_EscapeControlsCString(char const* in, size_t inLength,
         *qtr++ = '\\';
         *qtr = 'r';
         break;
+      
+      case '\t':
+        *qtr++ = '\\';
+        *qtr = 't';
+        break;
 
       default:
         n = (uint8_t)(*ptr);
 
-        if (n < 32 || n > 127) {
+        if (n < 32) {
           uint8_t n1 = n >> 4;
           uint8_t n2 = n & 0x0F;
 
```
I just have to find out the rationale behind escaping printable non-ASCII characters.

**Comment by jsteemann** [comment]
@fceller : should we apply the patch?

**Comment by fceller** [comment]
@jsteemann yes

**Comment by jsteemann** [comment]
Fixed in 3.0.3

**Comment by ilearnio** [comment]
Thanks!
